annotations/views/repository_views.py

```diff
@@ -313,12 +313,37 @@ def repository_collection_texts(request, repository_id, group_id, group_collecti
     for collection_item in texts['group']:
         if collection_item['key'] == group_collection_id:
             collection_name = collection_item['name']
+            break
 
     # retrieve items per page from settings and calculate pagination metadata from util function
     items_per_page = settings.PAGINATION_PAGE_SIZE
     pagination = get_pagination_metadata(total_items=texts.get('total_items'), page=page, items_per_page=items_per_page)
 
+    subcollections = []
+    try:
+        subcollections = manager.get_subcollections(group_id, group_collection_id)
+        # Sort subcollections alphabetically
+        subcollections = sorted(
+            subcollections, 
+            key=lambda x: x['name'].lower()
+        )
+    except CitesphereAPIError as e:
+        print(traceback.format_exc())
+        return render(request, 'annotations/repository_ioerror.html', {'error': str(e)}, status=500)
+    except Exception as e:
+        print(traceback.format_exc())
+        return render(request, 'annotations/repository_ioerror.html', {'error': 'An unexpected error occurred'}, status=500)
     project_id = request.GET.get('project_id')
+    
+    if not collection_name:
+        parent_collection_key = request.GET.get('parent_collection_key')
+        if parent_collection_key:
+            parent_data = manager.get_subcollections(group_id, parent_collection_key)
+            for collection_item in parent_data:
+                if collection_item['key'] == group_collection_id:
+                    collection_name = collection_item['name']
+                    break
```

Comment:
this loop can be more pythonic, something like
```
next(collection_item['name'] for collection_item in parent_data if collection_item['key'] == group_collection_id)
```
(haven't tried to run this, but I imagine something like that would work and makes it a lot shorter.